### PR TITLE
Implement a nice splash screen for the web driver.

### LINKED ISSUE #2756: Improve initial web page for the web driver
### Is your feature request related to a problem? Please describe:
Currently their is no user feedback during the initialization of the web driver. It would be good to provide user feedback during the progress of the initialization.

### Is it possible to construct a solution with the existing API?
There should be no need for any new API.

### Describe the solution you'd like to see:
The index.html provided by fyne could:
- detect that the browser does not have javascript support, display warning if not
- detect that the browser support WebGL, display warning if not
- detect that the browser support webassembly, fallback to gopherjs backend
- let the user know that the webassembly is being downloaded

### PR DESCRIPTION
**Pull request by Bluebugs**
### Description:
This nicely improve the splash screen when loading a Fyne app on the web. It use the application logo, its name and version to build this splash screen.

Fixes #2756

### Checklist:
- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:
- [x] Updated the vendor folder (using `go mod vendor`).

### CODE REVIEW

**Comment by andydotxyz** [comment]
Looks like this broke the web package tests...

File: cmd/fyne/internal/commands/package.go [hunk 1]
```diff
@@ -338,8 +338,16 @@ func (p *Packager) validate() error {
 		_, _ = fmt.Fprint(os.Stderr, "Parameter -executable is ignored for mobile builds.\n")
 	}
 
-	if p.name == "" || p.os == "wasm" || p.os == "gopherjs" || p.os == "web" {
-		p.name = exeName
+	if p.os == "wasm" || p.os == "gopherjs" || p.os == "web" {
+		if p.name == "" {
```
Comment (andydotxyz) [inline cmd/fyne/internal/commands/package.go:342]: I think this logic has broken the case where there is no name set, but the OS is not one of the web targets...
It should default to the exeName as it used to.

**Review by andydotxyz — CHANGES_REQUESTED** [review]
The addition looks great, but I think a logic re-order has broken a condition for non-web code.

File: cmd/fyne/internal/commands/package.go [hunk 2]
```diff
@@ -376,10 +376,6 @@ func calculateExeName(sourceDir, os string) string {
 
 	if os == "windows" {
 		exeName = exeName + ".exe"
-	} else if os == "wasm" {
```
Comment (andydotxyz) [inline cmd/fyne/internal/commands/package.go:379]: is it OK that this code is removed?

Comment (Bluebugs) [reply cmd/fyne/internal/commands/package.go:379]: Yes, I have moved the logic to add the extension in package-web.go where it is actually making more sense as it match the case where we are generating two executable and need to call build appropriately.

**Review by andydotxyz — APPROVED** [review]
Cool, thanks

**Pull request merged**